iceoryx_posh/include/iceoryx_posh/popo/wait_set.hpp

```diff
@@ -26,129 +31,96 @@ class Condition;
 
 enum class WaitSetError : uint8_t
 {
-    CONDITION_VECTOR_OVERFLOW,
+    TRIGGER_VECTOR_OVERFLOW,
+    TRIGGER_ALREADY_ACQUIRED,
 };
 
-/// @brief Logical disjunction of a certain number of Conditions
+/// @brief Logical disjunction of a certain number of Triggers
 ///
-/// The WaitSet stores Conditions and allows the user to wait till those set of Conditions become true. It works over
-/// process borders. With the creation of a WaitSet it requests a condition variable from RouDi and destroys it with the
-/// destructor. Hence the lifetime of the condition variable is bound to the lifetime of the WaitSet. Before destroying
-/// a Condition object the user has to ensure that it was detached from the WaitSet before.
-///
-///
-/// @code
-///
-/// @todo Add gateway use-case to example code
-///
-/// // Create Waitset
-/// WaitSet myWaitSet();
-///
-/// Subscriber mySubscriber1({"Radar", "FrontRight", "Counter"});
-/// Subscriber mySubscriber2({"Radar", "FrontLeft", "Counter"});
-/// GuardCondition myGuardCond;
-///
-///
-/// myWaitSet.attachCondition(mySubscriber1);
-/// myWaitSet.attachCondition(mySubscriber2);
-/// myWaitSet.attachCondition(myGuardCond);
-///
-/// std::thread someOtherThread{[&](){
-///     while(true)
-///     {
-///       if(IWantToWakeUpTheWaitSet)
-///       {
-///           // Let the WaitSet return
-///           myGuardCond.setTrigger();
-///       }
-///       // Don't forget to reset the trigger
-///       myGuardCond.resetTrigger();
-///     }
-/// }};
-///
-/// while (true)
-/// {
-///     // Wait till new data has arrived (any condition has become true)
-///     auto vectorOfFulfilledConditions = myWaitSet.wait();
-///
-///     for(auto& element : vectorOfFulfilledConditions)
-///     {
-///         if(element == &mySubscriber1)
-///         {
-///             // Subscriber1 has received new data
-///             ChunkHeader myData;
-///             mySubscriber1.tryGetChunk(myData);
-///             doSomeThingWithTheNewData(myData);
-///         }
-///     }
-/// }
-/// someOtherThread.join();
-///
-/// myWaitSet.detachCondition(guardCond);
-/// myWaitSet.detachCondition(mySubscriber2);
-/// myWaitSet.detachCondition(mySubscriber1);
-///
-/// @endcode
+/// The WaitSet stores Triggers and allows the user to wait till one or more of those Triggers are triggered. It works
+/// over process borders. With the creation of a WaitSet it requests a condition variable from RouDi and destroys it
+/// with the destructor. Hence the lifetime of the condition variable is bound to the lifetime of the WaitSet.
 class WaitSet
 {
   public:
-    using ConditionVector = cxx::vector<Condition*, MAX_NUMBER_OF_CONDITIONS_PER_WAITSET>;
+    using TriggerVector = cxx::vector<Trigger, MAX_NUMBER_OF_TRIGGERS_PER_WAITSET>;
```

Comment:
shall we make the waitset a template with `MAX_NUMBER_OF_TRIGGERS_PER_WAITSET` as default

Reply:
I would do this when we have a solution for the c binding.